packages/beacon-node/src/network/core/types.ts

```diff
@@ -87,6 +87,9 @@ export type NetworkWorkerData = {
  * API exposed by the libp2p worker
  */
 export type NetworkWorkerApi = INetworkCorePublic & {
+  // To satisfy the constraint of `ModuleThread` type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [string: string]: (...args: any[]) => Promise<any> | any;
```

Comment:
don't we lose type safety when consuming the API? Type casting in some other place might be preferable

Reply:
No we still have type safety, just that we don't have to do `as unknown as ` when using in threads library.